tests/phpunit/includes/test-wc-gateway-amazon-payments-advanced.php

```diff
@@ -0,0 +1,147 @@
+<?php
+/**
+ * Test cases for WC_Gateway_Amazon_Payments_Advanced.
+ *
+ * @package WC_Gateway_Amazon_Pay/Tests
+ */
+
+declare(strict_types=1);
+
+/**
+ * WC_Gateway_Amazon_Payments_Advanced_Test tests functionalities in WC_Gateway_Amazon_Payments_Advanced.
+ */
+class WC_Gateway_Amazon_Payments_Advanced_Test extends WP_UnitTestCase {
+
+	/**
+	 * Store an instance of our Gateway being tested.
+	 *
+	 * @var ?WC_Gateway_Amazon_Payments_Advanced
+	 */
+	protected static $gateway;
+
+	/**
+	 * Set up our DB before tests.
+	 *
+	 * @return void
+	 */
+	public static function set_up_before_class() : void {
+		parent::set_up_before_class();
+		self::$gateway = new WC_Gateway_Amazon_Payments_Advanced();
+		update_option( 'woocommerce_currency', 'EUR' );
+		update_option( 'woocommerce_amazon_payments_new_install', WC_AMAZON_PAY_VERSION );
+		update_option( 'amazon_api_version', 'V2' );
+	}
+
+	/**
+	 * Restore the DB's state after tests.
+	 *
+	 * @return void
+	 */
+	public static function tear_down_after_class() : void {
+		parent::tear_down_after_class();
```

Comment:
Similar to destructors, the parent call is a better call as the last statement.

Reply:
yea makes sense, done!